zebra-state/src/sled_state.rs

```diff
@@ -181,12 +184,29 @@ impl FinalizedState {
                 .coinbase_height()
                 .expect("valid blocks must have a height");
             self.commit_finalized(queued_block);
-            metrics::counter!("state.committed.block.count", 1);
-            metrics::gauge!("state.committed.block.height", height.0 as _);
+            metrics::counter!("state.finalized.committed.block.count", 1);
+            metrics::gauge!("state.finalized.committed.block.height", height.0 as _);
+        }
+
+        if self.queued_by_prev_hash.is_empty() {
+            self.max_queued_height = None;
+            // use -1 as a sentinel value for "None", because 0 is a valid height
+            metrics::gauge!("state.finalized.queued.max.height", -1,);
+        } else {
+            self.max_queued_height = self
+                .max_queued_height
+                .filter(|old_max| old_max > &height)
+                .or(Some(height));
+            metrics::gauge!(
+                "state.finalized.queued.max.height",
+                self.max_queued_height
+                    .expect("previous statement ensures that height is valid")
+                    .0 as _,
+            );
         }
```

Comment:
I think this is a bit easier to read if we just avoid `Option`s, how do you feel about this version (assuming its equivalent).

```suggestion
        if self.queued_by_prev_hash.is_empty() {
            // use -1 as a sentinel value for "None", because 0 is a valid height
            self.max_queued_height = -1;
        } else {
            self.max_queued_height = std::cmp::max(self.max_queued_height, height.0);
        }

        metrics::gauge!("state.finalized.queued.max.height", self.max_queued_height);
```